### ISSUE TITLE
battery status support for devices

### ISSUE THREAD

**Issue by draconicfae** [godotengine/godot#29791]
I have a device (bluetooth virtual reality controller) which, in addition to having buttons, touchpad, gyroscope, magnetometer, and accelerometer, also makes its battery status available.  I see that Godot support for battery status for the system itself (eg laptop) already exists, but it would be great if devices that connect to Godot could also report how much battery they have left.  I'm willing to code this for x11 and probably also Android if I can get guidance as to what sort of approach would be most likely to be accepted.  For instance, would it make sense to mirror the information that SDL provides for the system itself (a powerstate as well as percent and minutes remaining when available) or would that create confusion?  Further, for an x11 implementation for devices via udev, what I've found is KEY_BATTERY (line 306 at https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h ) but I'm not sure whether or not that's connected to a standard way to convey battery information or if it just means "user has pressed a key that semantically means some sort of battery thing should happen".  

Basically, I'd love to write this feature but would appreciate some guidance so that I code the right thing.

**Comment by groud** [comment]
I don't think this deserves to be in the core, likely more something to be made as a gdnative plugin (It pretty uncommon for games to need such information).

**Comment by KoBeWi** [comment]
Isn't it the same as #8770?

**Comment by groud** [comment]
No, I had checked too. This issue is for devices power (like joypads on battery), not the laptop's one. They are a little bit related but not that much in fact.

**Comment by alex-evelyn** [comment]
I am also trying to figure out how to implement some missing features for my controller. In my case, it is color led and fixing haptic feedback for the DS4 (on OSX).

Once I figure out how to modify these values, I want to incorporate them with Input (preferably as a gdnative module so others can use it).

**Comment by mhilbrunner** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
